### PR TITLE
clean up admin commands

### PR DESCRIPTION
**Pull request by deads2k**
This properly segregate master cert generation from node cert generation.  It eliminates unnecessary commands and starts properly passing command names and io.Writer

@liggitt

### CODE REVIEW

File: pkg/cmd/server/admin/create_bootstrappolicy_file.go [hunk 1]
```diff
@@ -32,15 +33,15 @@ type CreateBootstrapPolicyFileOptions struct {
 	OpenShiftSharedResourcesNamespace string
 }
 
-func NewCommandCreateBootstrapPolicyFile() *cobra.Command {
+func NewCommandCreateBootstrapPolicyFile(commandName string, out io.Writer) *cobra.Command {
 	options := &CreateBootstrapPolicyFileOptions{}
```
Comment (smarterclayton) [inline pkg/cmd/server/admin/create_bootstrappolicy_file.go]: Pass in fullName, name like the other cli commands

Comment (deads2k) [inline pkg/cmd/server/admin/create_bootstrappolicy_file.go]: > Pass in fullName, name like the other cli commands

Done.

File: pkg/cmd/server/admin/create_bootstrappolicy_file.go [hunk 2]
```diff
@@ -32,15 +33,15 @@ type CreateBootstrapPolicyFileOptions struct {
 	OpenShiftSharedResourcesNamespace string
 }
 
-func NewCommandCreateBootstrapPolicyFile() *cobra.Command {
+func NewCommandCreateBootstrapPolicyFile(commandName string, baseName string, out io.Writer) *cobra.Command {
```
Comment (smarterclayton) [inline pkg/cmd/server/admin/create_bootstrappolicy_file.go]: Can you call them the same things the other ones do?  `fullName` and `name`.  And yes, the others are wrong if they're not like that.

Comment (deads2k) [inline pkg/cmd/server/admin/create_bootstrappolicy_file.go]: Seems like we'd want to pass name and baseName as opposed to name and fullName.  Why would we want fullName instead?

Comment (smarterclayton) [inline pkg/cmd/server/admin/create_bootstrappolicy_file.go]: fullName is the name someone access you as in a larger context.  Name is your name.  

You are David.  Your full name is David Eads.

----- Original Message -----

Comment (deads2k) [inline pkg/cmd/server/admin/create_bootstrappolicy_file.go]: And if you were passing information to an API, you wouldn't want to pass FullName, you'd  pass FirstName LastName (or GivenName SurName).  

Should args be: "create-master-certs", "openshift admin create-master-certs" or "create-master-certs", "openshift admin".  I vote for the latter and some commands are taking that, but calling the second arg "fullName" in spite of it being a "baseName".

Comment (smarterclayton) [inline pkg/cmd/server/admin/create_bootstrappolicy_file.go]: > On Mar 26, 2015, at 10:40 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/cmd/server/admin/create_bootstrappolicy_file.go:
> 
> > @@ -32,15 +33,15 @@ type CreateBootstrapPolicyFileOptions struct {
> >  	OpenShiftSharedResourcesNamespace string
> >  }
> >  
> > -func NewCommandCreateBootstrapPolicyFile() *cobra.Command {
> > +func NewCommandCreateBootstrapPolicyFile(commandName string, baseName string, out io.Writer) *cobra.Command {
> And if you were passing information to an API, you wouldn't want to pass FullName, you'd pass FirstName LastName (or GivenName SurName).
> 
> Should args be: "create-master-certs", "openshift admin create-master-certs" or "create-master-certs", "openshift admin". I vote for the latter and some commands are taking that, but calling the second arg "fullName" in spite of it being a "baseName".
> 
If you do that, every command has to do string manipulation.  Stick to fullName name, but if you have a better naming scheme, do so.

File: hack/test-cmd.sh [hunk 3]
```diff
@@ -86,8 +86,8 @@ do
     SERVER_HOSTNAME_LIST="${SERVER_HOSTNAME_LIST},${IP_ADDRESS}"
 done <<< "${ALL_IP_ADDRESSES}"
 
-# Create certificates
-openshift admin create-all-certs --overwrite=false --cert-dir="${CERT_DIR}" --hostnames="${SERVER_HOSTNAME_LIST}" --nodes="${API_HOST}" --master="${MASTER_ADDR}" --public-master="${API_SCHEME}://${PUBLIC_MASTER_HOST}"
+openshift admin create-master-certs --overwrite=false --cert-dir="${CERT_DIR}" --hostnames="${SERVER_HOSTNAME_LIST}" --master="${MASTER_ADDR}" --public-master="${API_SCHEME}://${PUBLIC_MASTER_HOST}"
+openshift admin create-node-config --listen="https://0.0.0.0:10250" --node-dir="${CERT_DIR}/node-${API_HOST}" --node="${API_HOST}" --hostnames="${SERVER_HOSTNAME_LIST}" --master="${MASTER_ADDR}" --certificate-authority="${CERT_DIR}/ca/cert.crt" --signer-cert="${CERT_DIR}/ca/cert.crt" --signer-key="${CERT_DIR}/ca/key.key" --signer-serial="${CERT_DIR}/ca/serial.txt"
```
Comment (liggitt) [inline hack/test-cmd.sh]: create-node-config shouldn't listen on https yet

File: hack/test-cmd.sh [hunk 4]
```diff
@@ -86,8 +86,8 @@ do
...
 
 # Start openshift
 OPENSHIFT_ON_PANIC=crash openshift start --master="${API_SCHEME}://${API_HOST}:${API_PORT}" --listen="${API_SCHEME}://${API_HOST}:${API_PORT}" --hostname="${API_HOST}" --volume-dir="${VOLUME_DIR}" --cert-dir="${CERT_DIR}" --etcd-dir="${ETCD_DATA_DIR}" 1>&2 &
```
Comment (liggitt) [inline hack/test-cmd.sh]: If we want to make sure our prep commands did their job, add `--create-certs=false` here?

File: hack/test-extended.sh [hunk 5]
```diff
@@ -56,15 +56,26 @@ start_server() {
...
+  --hostnames="${SERVER_HOSTNAME_LIST}" \
+  --master="https://${OS_MASTER_ADDR}" \
+  --certificate-authority="${CERT_DIR}/ca/cert.crt" \
+  --signer-cert="${CERT_DIR}/ca/cert.crt" \
+  --signer-key="${CERT_DIR}/ca/key.key" \
+  --signer-serial="${CERT_DIR}/ca/serial.txt"
+
   echo "[INFO] Starting OpenShift server"
   sudo env "PATH=${PATH}" openshift start \
     --listen="https://0.0.0.0:${OS_MASTER_PORT}" \
```
Comment (liggitt) [inline hack/test-extended.sh]: add `--create-certs=false`?

File: pkg/cmd/server/admin/create_comands.go [hunk 6]
```diff
@@ -1,10 +1,12 @@
...
 import (
+	"io"
+
 	"github.com/spf13/cobra"
 )
 
-func NewCommandAdmin() *cobra.Command {
+func NewCommandAdmin(commandName string, baseName string, out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "admin",
```
Comment (liggitt) [inline pkg/cmd/server/admin/create_comands.go]: `commandName`

File: hack/test-extended.sh (hunk 5 above)
Comment (liggitt) [inline hack/test-extended.sh]: nm, already existed below

File: hack/test-cmd.sh (hunk 3 above)
Comment (liggitt) [inline hack/test-cmd.sh]: nm, I'll fix that up in my https PR

**Comment by liggitt** [comment]
Add --create-certs=false to test-cmd, use `commandName` in `create_commands.go`, then LGTM

**Comment by liggitt** [comment]
I'll wait for this to go in on Monday before merging the https node PR

**Comment by deads2k** [comment]
@liggitt  Comments seem to disappearing, but the create-node-config uses an https --listen because the server uses that by default and causes an all-in-one server without a specified config to expect to find node server certs.  I think that splitting those parameters is outside the scope of what I'm trying to do here, so the https --listen will create the certs as expected, but still serve insecurely

**Comment by deads2k** [comment]
[merge]

**Comment by liggitt** [comment]
asset fix merged, re[merge]

**Pull request merged**